### PR TITLE
CDI guide - mention Jandex maven plugin

### LINKED ISSUE #2945: Tell how to create a Jandex index in dependencies for multi-modules projects
We tell people they need to generate a Jandex index https://quarkus.io/guides/cdi-reference#bean_discovery
But we don't tell them how? It would be nice to point to the doc or simply tell them right away.

Likewise telling where to create the beans.xml `resources/META-INF/beans.xml` would be helpful, I saw a user making the mistake.

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #2945

### CODE REVIEW

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 1]
```diff
@@ -25,21 +25,48 @@ NOTE: Most of the existing CDI code should work just fine but there are some sma
...
+* dependencies that contain a Jandex index (`META-INF/jandex.idx`),
 * and Quarkus integration code.
 
 Bean classes that don't have a http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#bean_defining_annotations[bean defining annotation, window="_blank"] are not discovered.
 This behavior is defined by CDI.
 But producer methods and fields and observer methods are discovered even if the declaring class is not annotated with a bean defining annotation (this behavior is different to what is defined in CDI).
 In fact, the declaring bean classes are considered annotated with `@Dependent`. 
 
-NOTE: Quarkus extension may declare additional discovery rules. For example `@Scheduled` business methods are registered even if the declaring class is not annotated with a bean defining annotation.
+NOTE: Quarkus extension may declare additional discovery rules. For example, `@Scheduled` business methods are registered even if the declaring class is not annotated with a bean defining annotation.
```
Comment (gsmet) [inline docs/src/main/asciidoc/cdi-reference.adoc]:
```suggestion
NOTE: Quarkus extensions may declare additional discovery rules. For example, `@Scheduled` business methods are registered even if the declaring class is not annotated with a bean defining annotation.
```
(or `A Quarkus extension`)
I know it was preexisting but better fix it.

**Review by gsmet — DISMISSED** [review]
@mkouba looks good. I added a minor suggestion about an existing wording, could you have a look?

**Comment by machi1990** [comment]
@gsmet @mkouba do we need to mention the indexing third party dependencies?

**Comment by gsmet** [comment]
@machi1990 yeah, mentioning `quarkus.index-dependency.*` would make sense.

@mkouba WDYT?

**Comment by mkouba** [comment]
@gsmet @machi1990 Done ;-)

**Comment by machi1990** [comment]
@mkouba LGTM.

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 2]
```diff
@@ -25,22 +25,67 @@ NOTE: Most of the existing CDI code should work just fine but there are some sma
...
+However, Quarkus is using a *simplified bean discovery*.
+There is only one bean archive with https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#default_bean_discovery[`annotated` bean discovery mode, window="_blank"] and no visibility boundaries.
 
 The bean archive is synthesized from:
 
-* the application,
-* application dependencies that contain a `beans.xml` descriptor or a generated Jandex index (`META-INF/jandex.idx`),
+* the application classes,
+* dependencies that contain a `beans.xml` descriptor (content is ignored),
+* dependencies that contain a Jandex index (`META-INF/jandex.idx`),
```
Comment (gsmet) [inline docs/src/main/asciidoc/cdi-reference.adoc]: Shouldn’t we add the quarkus.index-dependency to this list too?

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 3]
```diff
@@ -25,22 +25,68 @@ NOTE: Most of the existing CDI code should work just fine but there are some sma
 == Bean Discovery
 
 Bean discovery in CDI is a complex process which involves legacy deployment structures and accessibility requirements of the underlying module architecture.
-Quarkus is using a simplified bean discovery.
-There is only one bean archive with `annotated` bean discovery mode and no visibility boundaries.
+However, Quarkus is using a *simplified bean discovery*.
```
Comment (geoand) [inline docs/src/main/asciidoc/cdi-reference.adoc]: I would rather have it say `Quarkus uses`, but let's not hold up the PR for something so simple

**Review by geoand — APPROVED** [review]

**Pull request merged**